### ISSUE TITLE
Turret collection mod

### ISSUE THREAD

**Issue by ManiacDK** [AthenaSulisMinerva/CombatExtendedFastTrack#33]
Would it be possible to get a patch to get the above mentioned mod to be compatible with CE?
I am still very new to modding, and know next to no coding, so I don't really want to go deep diving and end up ruining more things that fixing.
Thanks in advance :)

**Comment by ManiacDK** [comment]
Also I just started using FastTrack instead for my new playthrough, and I am getting some XML errors,
XML error: Duplicate XML node name MeleeParryChance or (MeleeDodgeChance, MeleeCritChance) in this XML block: <statBases><MoveSpeed>3.35</MoveSpeed><ComfyTemperatureMin>-16</ComfyTemperatureMin><MarketValue>225</MarketValue><MeleeDodgeChance>0.17</MeleeDodgeChance><MeleeCritChance>0.1</MeleeCritChance><MeleeParryChance>0.05</MeleeParryChance><MeleeDodgeChance>0.17</MeleeDodgeChance><MeleeCritChance>0.1</MeleeCritChance><MeleeParryChance>0.05</MeleeParryChance></statBases>, there is a lot more but I am not sure how much I should be including,
Any help would be nice :)

**Comment by ManiacDK** [comment]
Also when using CE:FT, I can no longer craft a lot of the weapons.
Is there any way to fix this.

**Comment by AthenaSulisMinerva** [comment]
@sumghai - Sorry dude, here's another one for you. But no rush, so no milestone for now.

@ManiacDK - Please open separate issues for each of your questions - I deleted your other ones because they had nothing to do with your first request for mod compatibility. The GitHub issue tracker is not a forums or chat thread.

**Comment by Taranchuk** [comment]
Actually Turret collection mod made by eatKenny:
https://ludeon.com/forums/index.php?topic=6895.0
leafzxg is just a steam uploader.

**Comment by sumghai** [comment]
Many of these turrets require new ammo definitions, which I have now added.

I'll also need to create new textures of some of them, like the rockets and the tank cannon shells.

**Comment by sumghai** [comment]
This took an entire day's worth of painstaking debugging, but the patch is pretty much working now.

Caveats (please include in release notes, @AthenaSulisMinerva )
- Pre-charged Mini-Turret has been disabled, as CE and CE:FT current don't support spawning turrets preloaded with ammo; cursory examination of the relevant C# source code suggest that the upstream CE devs nerfed that feature, for reasons unknown to us
- TC's original ammo have all been disabled, and replaced with CE:FT's own version (so that the ammo can be reused by other third-party mods)
- TC's original research projects have all been replaced with `Heavy Weapons` and `Heavy Auto Turrets` from CE

All that's left to do is the ammo and projectile artwork.

**Comment by sumghai** [comment]
Only the following ammo and projectiles need new textures:

* 130mm Type 63 Rocket
* 120mm Cannon Shell
* 28cm Spgr. Cannon Shell

All other ammo can use existing CE artwork.

**Comment by sumghai** [comment]
New ammo graphics are done.

The 28cm Spgr. shells don't need unique graphics, since I just found out that the shells and cartridges are loaded separately (much like the powder bags used for other naval artillery). I've also rebalanced the market value and crafting stats accordingly.

One last thing to do is to rework the M74 rocket ammo so that there are two different projectile definitions for the one ammo, depending on usage:

* The standard version, with `flyOverhead=false` for mods that add the M202 FLASH rocket launcher
* A version specific to Turret Collection, with `flyOverhead=true` to support non-line-of-sight rocket artillery launch behaviour; I may even define this as within the Turret Collection patch rather than in the main CE:FT ammo library
